Firestore/Source/Remote/FSTRemoteStore.mm

```diff
@@ -315,10 +316,9 @@ - (void)watchStreamDidChange:(FSTWatchChange *)change
     // older than a previous snapshot we've processed (can happen after we resume a target
     // using a resume token).
     [self.accumulatedChanges addObject:change];
-    FSTAssert(snapshotVersion, @"snapshotVersion must not be nil.");
-    if ([snapshotVersion isEqual:[FSTSnapshotVersion noVersion]] ||
-        [snapshotVersion compare:[self.localStore lastRemoteSnapshotVersion]] ==
-            NSOrderedAscending) {
+    if (snapshotVersion == SnapshotVersion::None() ||
+        snapshotVersion <
+            static_cast<SnapshotVersion>([self.localStore lastRemoteSnapshotVersion])) {
```

Comment:
Is this safe? FSTLocalStore::lastRemoteSnapshotVersion still returns a FSTSnapshotVersion*, doesn't it?

Reply:
SnapshotVersion defines conversions that make this work:

https://github.com/firebase/firebase-ios-sdk/blob/master/Firestore/core/src/firebase/firestore/model/snapshot_version.h#L47

What I'm not clear on is why this conversion did not happen implicitly. An alternative that makes it clearer that we're invoking this conversion is to just use `SnapshotVersion{[self.localStore lastRemoteSnapshotVersion]}`.

Reply:
If it can't be made to happen implicitly, I like wilhuff@'s suggestion. (`static_cast` makes me all sorts of nervous.)

Reply:
The reason for explicit conversion is due to ambiguity when compare `SnapshotVersion` and `FSTSnapshotVersion`, where there is bi-directional implicit conversion.

Reply:
Ah, so snapshotVersion could be converted from a SnapshotVersion to a FSTSnapshotVersion while the return from lastRemoteSnapshotVersion could be converted from a FSTSnapshotVersion to a SnapshotVersion and the compiler doesn't know which to pick?

I guess static_cast is a perfectly reasonable thing to do here. I think I still prefer Gil's suggestion, but am ok with either.